mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulCreateTournamentScreen.kt

```diff
@@ -0,0 +1,134 @@
+    cancelClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+  val scope = rememberCoroutineScope()
+
+  val tournamentFacade = LocalTournamentFacade.current
+  val strings = LocalLocalizedStrings.current
+
+  val state =
+      remember(user, navigateToTournament, cancelClick, scope, tournamentFacade) {
+        ActualCreateTournamentScreenState(
+            user = user,
+            navigateToTournament = navigateToTournament,
+            cancelClick = cancelClick,
+            tournamentFacade = tournamentFacade,
+            strings = strings,
+            scope = scope,
+        )
+      }
+  CreateDialog(
+      state = state,
+      modifier = modifier,
+      poolSizeKey = { it.name },
+      eliminationRoundKey = { it.name },
+  )
+}
+
+class ActualCreateTournamentScreenState(
+    val user: AuthenticatedUser,
+    val navigateToTournament: (reference: TournamentReference) -> Unit,
+    val cancelClick: () -> Unit,
+    val tournamentFacade: TournamentFacade,
+    val strings: LocalizedStrings,
+    val scope: CoroutineScope,
+) : CreateDialogState<IntChoice, IntChoice> {
+
+  override var name: String by mutableStateOf("")
+
+  override fun onBestOfClick(count: Int) {
+    bestOf = count
+  }
+  override val bestOfChoices: List<Int> = listOf(1, 3, 5)
+  override var bestOf: Int? by mutableStateOf(null)
+    private set
+
+  override var maximumPlayerCount: String by mutableStateOf("")
+
+  override fun onPoolSizeClick(poolSize: IntChoice) {
+    this.poolSize = poolSize
+  }
+  override val poolSizeChoices: List<IntChoice>
+    get() =
+        listOf(IntChoice(name = strings.tournamentsCreateQualifierSize0, value = 0)) +
+            (2..(maximumPlayerCount.toIntOrNull() ?: 0)).map {
+              IntChoice(name = strings.tournamentsCreateQualifierSizeN(it.toString()), value = it)
+            }
+
+  override var poolSize: IntChoice? by mutableStateOf(null)
+    private set
+
+  override fun onEliminationRoundClick(eliminationRound: IntChoice) {
+    this.eliminationRound = eliminationRound
+  }
+  override val eliminationRoundChoices: List<IntChoice>
+    get() {
+      val players = (maximumPlayerCount.toIntOrNull() ?: 0) / 2
+      val depth = log2(players.toDouble()).toInt()
+
+      fun intPow(base: Int, exponent: Int): Int {
+        return base.toDouble().pow(exponent).toInt()
+      }
+
+      return listOf(IntChoice(name = strings.tournamentsCreateElimDepthFinal, value = 1)) +
+          (1..depth).map {
+            IntChoice(
+                name = strings.tournamentsCreateElimDemomN(intPow(2, it).toString()),
+                value = it + 1,
+            )
+          }
+    }
+  override var eliminationRound: IntChoice? by mutableStateOf(null)
+    private set
+
+  override val confirmEnabled: Boolean
+    get() =
+        name.isNotBlank() &&
+            bestOf != null &&
+            maximumPlayerCount.isDigitsOnly() &&
+            poolSize != null &&
+            eliminationRound != null
```

Comment:
Question : is there no confirmation that the currently selected values are correct ? What if the user inputs some correct values, and then changes the player count ?

Reply:
You're right. Fixed in [91e5118](https://github.com/epfl-SDP/android/pull/336/commits/91e51180395a60d4211803e7d03d421ef259cd5f)